dapp/src/node/mod.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    collections::HashMap,
+    net::SocketAddr,
+    sync::{Arc, RwLock},
+};
+
+use axum::{
+    extract::{rejection::JsonRejection, Extension, Json},
+    handler::post,
+    AddExtensionLayer, Router, Server,
+};
+use ethers::{
+    core::k256::ecdsa::SigningKey,
+    prelude::{Address, Signer, Wallet, U256, U64},
+};
+use evm_adapters::Evm;
+
+mod blockchain;
+use blockchain::Blockchain;
+
+mod methods;
+use methods::{EthRequest, EthResponse};
+
+mod types;
+use types::{Error, JsonRpcRequest, JsonRpcResponse, ResponseContent};
+
+// TODO: impl builder style for node config
+pub struct NodeConfig {
+    chain_id: U64,
+    genesis_accounts: Vec<Wallet<SigningKey>>,
+    genesis_balance: U256,
+    accounts: HashMap<Address, Wallet<SigningKey>>,
+}
+
+impl NodeConfig {
+    pub fn new<U: Into<U64>>(
+        chain_id: U,
+        genesis_accounts: Vec<Wallet<SigningKey>>,
+        balance: U256,
+    ) -> Self {
+        let mut accounts = HashMap::with_capacity(genesis_accounts.len());
+        genesis_accounts.iter().cloned().for_each(|w| {
+            accounts.insert(w.address(), w);
+        });
+        Self { chain_id: chain_id.into(), genesis_accounts, genesis_balance: balance, accounts }
+    }
+}
+
+pub struct Node<E> {
+    evm: E,
+    config: NodeConfig,
+    blockchain: Blockchain,
+}
+
+pub type SharedNode<E> = Arc<RwLock<Node<E>>>;
+
+impl<E: Send + Sync + 'static> Node<E> {
```

Comment:
```suggestion
impl<E: Send + Sync + 'static> Node<E> {
```

nit: Move these to the `where` clause below